File: examples/python/ds/model/genre.py
from ds.model._reference import *
from quartz.error import Error
from quartz.util import Parse

#
# Entity
#

class Genre(GenreRef):

    def __init__(self, name=None, id=None):
        super().__init__(id)
        self.name = name

    def __str__(self):
        return Parse.string(self._name)

    def __repr__(self):
        ref = super().__repr__()
        name = Parse.string(self._name)
        return "{}(name:{})".format(ref, name)

    def _validateRef(self, x):
        if isinstance(x, int) or isinstance(x, GenreRef):
            return int(x)   

    @property
    def name(self):
        return self._name
    @name.setter
    def name(self, x):
        self._name = x

    def copy(self, x):
        if isinstance(x, Genre):
            super().copy(x)
            self.name = x.name
        return self

File: examples/php/ds/model/genre.php
<?php
namespace books\model;
require_once __DIR__ . '/../base.php';


class Genre extends \books\GenreRef {
    public $name;

    function __construct($name=null, $id=null) {
        parent::__construct($id);
        $this->name = $name;
    }

    public function __toString(): string {
        $ref = parent::__toString();
        $name = is_null($this->name) ? '∅' : $this->name;
        return "genre$ref(name:{$name})";
    }

    public function copy($x) {
        if ($x instanceof GenreRef) {
            parent::copy($x);
            $this->name = x.name;
        }
    }
}

?>